src/rust/engine/src/context.rs

```diff
@@ -79,11 +77,10 @@ impl Core {
     let mut remote_store_servers = remote_store_servers;
     remote_store_servers.shuffle(&mut rand::thread_rng());
 
-    let runtime = Resettable::new(|| {
+    let runtime =
       Arc::new(RwLock::new(Runtime::new().unwrap_or_else(|e| {
```

Comment:
`Runtime` is `Send` and `Sync` afaict, so the RwLock should no longer be necessary...? I wonder whether it ever was.

Reply:
...oh, sorry. Except we need mutable access to it for `spawn` and `block_on`. Nevermind.